dxr/plugins/core.py

```diff
@@ -306,6 +311,8 @@ def needles(self):
         extension = splitext(self.path)[1]
         if extension:
             yield 'ext', extension[1:]  # skip the period
+        if is_image(self.path):
```

Comment:
This should be `if self.contains_text() and is_image(self.path)`. Otherwise, we could end up base64ing a unicode string, and that might land us in weird places.